lib/main.dart

```diff
@@ -1,17 +1,24 @@
 import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
+import 'package:hive_flutter/adapters.dart';
 import 'package:myflix/core/dependencies/registry.dart';
 import 'package:myflix/core/models/movie_model.dart';
 import 'package:myflix/core/routes.dart';
 import 'package:myflix/core/utils/logger.dart';
 import 'package:myflix/features/details/presentation/view/movie_details_page.dart';
 import 'package:myflix/views/index_page.dart';
 
-void main() {
+void main() async {
+  await Hive.initFlutter();
+  WidgetsFlutterBinding.ensureInitialized();
+  Hive.registerAdapter(MovieAdapter());
+  await Hive.openBox<Movie>('watchListBox');
```

Comment:
Create a constants file and add this `'watchListBox'` as a constant, then reuse across your app to avoid spelling mistakes and something called 'magic strings'